src/main/java/com/booking/app/mapper/UkrainianPlacesMapper.java

```diff
@@ -5,11 +5,44 @@
 import org.mapstruct.Mapper;
 import org.mapstruct.Mapping;
 
+import java.util.List;
+
 @Mapper(componentModel = "spring")
 public interface UkrainianPlacesMapper {
 
-    @Mapping(target = "cityUa",source = "nameUa")
-    @Mapping(target = "cityEng",source = "nameEng")
+    @Mapping(target = "cityUa", source = "nameUa", ignore = true)
+    @Mapping(target = "cityEng", source = "nameEng", ignore = true)
+    @Mapping(target = "siteLanguage", ignore = true)
+    @Mapping(target = "country", source = "country")
     CitiesDTO toCitiesDTO(UkrainianPlaces ukrainianPlaces);
 
+    default List<CitiesDTO> toCitiesDTOList(List<UkrainianPlaces> ukrainianPlacesList, String inputLanguage, String siteLanguage) {
+
+        return ukrainianPlacesList.stream()
+                .map(place -> {
+                    CitiesDTO citiesDTO = toCitiesDTO(place);
+                    citiesDTO.setSiteLanguage(siteLanguage);
+                    if (siteLanguage.contains("ua")) {
+                        if (inputLanguage.equals("ua")) {
+                            citiesDTO.setCityUa(place.getNameUa());
+                        }
+                        if (inputLanguage.equals("eng")) {
+                            citiesDTO.setCityEng(place.getNameEng());
+                            citiesDTO.setCityUa(place.getNameUa());
+                        }
+                    }
+                    if (siteLanguage.contains("eng")) {
+                        if (inputLanguage.equals("eng")) {
+                            citiesDTO.setCityEng(place.getNameEng());
+                        }
+                        if (inputLanguage.equals("ua")) {
+                            citiesDTO.setCityEng(place.getNameEng());
+                            citiesDTO.setCityUa(place.getNameUa());
+                        }
+                    }
+                    return citiesDTO;
```

Comment:
can you use switch case here? like in this blog - https://blog.udemy.com/java-switch-statement/?utm_source=adwords&utm_medium=udemyads&utm_campaign=DSA_Catchall_la.EN_cc.ROW&utm_content=deal4584&utm_term=_._ag_88010211481_._ad_535397282064_._kw__._de_c_._dm__._pl__._ti_aud-2268488108799:dsa-449490803887_._li_1012839_._pd__._&matchtype=&gad_source=1&gclid=Cj0KCQiAqsitBhDlARIsAGMR1Rj6RXSuRi9XLPpoad87ldqQ2g7P0PIgd5CyYl7SFaPBTAqSDyq2cbsaAgtpEALw_wcB

Reply:
this is necessary for future adding new languages

Reply:
you can do changes and merge

Reply:
and it is great to see checks passed!!!